engine/src/main/java/org/hibernate/validator/internal/engine/valuecontext/ValueContexts.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.valuecontext;
+
+import org.hibernate.validator.internal.engine.path.PathImpl;
+import org.hibernate.validator.internal.metadata.BeanMetaDataManager;
+import org.hibernate.validator.internal.metadata.aggregated.BeanMetaData;
+import org.hibernate.validator.internal.metadata.facets.Validatable;
+import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
+
+/**
+ * @author Marko Bekhta
+ */
+public final class ValueContexts<T, V> {
+
+	private ValueContexts() {
+	}
+
+	public static <T, V> ValueContextWithBeanMetadata<T, V> getLocalExecutionContext(
+			BeanMetaDataManager beanMetaDataManager,
+			ExecutableParameterNameProvider parameterNameProvider,
+			T value,
+			Validatable validatable,
+			PathImpl propertyPath) {
+		@SuppressWarnings("unchecked")
+		Class<T> rootBeanType = (Class<T>) value.getClass();
+		return new ValueContextWithBeanMetadataImpl<>( parameterNameProvider, value, beanMetaDataManager.getBeanMetaData( rootBeanType ), validatable, propertyPath );
+	}
+
+	/**
+	 * Creates a value context for validataing parameters of an executable. Does not require a bean metadata
+	 * information.
+	 */
+	public static <T, V> ValueContext<T, V> getLocalExecutionContext(
```

Comment:
The comment above is untrue as it is also used for return value validation.

I think, in the end, once the above method is removed, we will end up with `getLocalExecutionContextForExecutable()`. Let's do a final check once the element above is removed before renaming.